ios/UARCTModule/UARCTEventEmitter.m

```diff
@@ -55,13 +55,18 @@ - (BOOL)sendEventWithName:(NSString *)eventName body:(id)body {
 }
 
 - (void)addListener:(NSString *)eventName {
-    self.listenerCount++;
-    if (self.listenerCount > 0) {
+   self.listenerCount++;
+    if (self.pendingEvents.count > 0) {
+        NSMutableArray *discardedItems = [NSMutableArray array];
+
         for (NSDictionary *event in self.pendingEvents) {
-            [self sendEventWithName:event[@"name"] body:event[@"body"]];
+            if ([event[@"name"] isEqualToString:eventName] &&
+                [self sendEventWithName:event[@"name"] body:event[@"body"]]) {
```

Comment:
can this ever return `NO` in this context ?

Reply:
It *should* always pass in this context but just in case it doesn't get delivered (e.g. the condition changes in the future) it is better not to remove it from the queue